test/AccessLock.t.sol

```diff
@@ -139,9 +140,28 @@ contract AccessLockTest is Test, Deployers {
         assertEq(manager.balanceOf(address(accessLockHook), currency1), amount);
     }
 
-    function test_beforeModifyPosition_take_succeedsWithAccessLock(uint128 amount) public {
+    function test_beforeRemoveLiquidity_mint_succeedsWithAccessLock(uint128 amount) public {
+        vm.assume(amount < uint128(type(int128).max));
+        uint256 balanceOfBefore1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        modifyPositionRouter.addLiquidity(key, LIQ_PARAMS, ZERO_BYTES);
+        modifyPositionRouter.removeLiquidity(key, LIQ_PARAMS, abi.encode(amount, AccessLockHook.LockAction.Mint));
+
+        uint256 balanceOfAfter0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+        uint256 balanceOfAfter1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        // The balance of our contract should be equal to our original balance because we added and removed our liquidity.
+        // TODO: Maybe due to a rounding issue, but balanceOfBefore0 == balanceOfAfter0 ?
+        assertTrue(balanceOfBefore0 - balanceOfAfter0 <= 1);
+        assertTrue(balanceOfBefore1 - balanceOfAfter1 - amount <= 1);
```

Comment:
Not sure why I had an off by one issue here?

Reply:
Ya should equal, that's werid